qiskit_experiments/curve_analysis/fit_models.py

```diff
@@ -0,0 +1,339 @@
+        instance = self.__class__(
+            fit_functions=self._fit_functions,
+            signatures=self._signatures,
+            fit_models=self._fit_models,
+        )
+
+        if self._fixed_params:
+            instance.bind_parameters(**self._fixed_params.copy())
+
+        return instance
+
+    def __repr__(self):
+        sigrepr = ", ".join(self.signature)
+        if self._fixed_params:
+            fixrepr = ", ".join(self._fixed_params.keys())
+            return f"{self.__class__.__name__}(x, {sigrepr}; @ Fixed {fixrepr})"
+        return f"{self.__class__.__name__}(x, {sigrepr})"
+
+
+class SingleFitFunction(FitModel):
+    r"""Fit model consisting of a single curve.
+
+    This model is created when only a single curve exists in the fit model.
+
+    .. math::
+
+        F(x, \Theta) = f(x, \vec{p})
+
+    The parameter :math:`\vec{p} = \Theta \cup \Theta_{\rm fix}` which is a union of
+    the fit parameters and the fixed parameters :math:`\Theta_{\rm fix}`.
+    The function :math:`f` is usually set by :attr:`SeriesDef.fit_func` which is
+    a standard python function.
+
+    .. seealso::
+
+        Class :class:`FitModel`.
+    """
+
+    def __call__(self, x: np.ndarray, *params) -> np.ndarray:
+        kwparams = dict(zip(self.signature, params))
+        kwparams.update(self._fixed_params)
+
+        return self._fit_functions[0](x, **{p: kwparams[p] for p in self._signatures[0]})
+
+
+class CompositeFitFunction(FitModel):
+    r"""Fit model consisting of multiple curves sharing fit parameters.
+
+    This model is created when multiple curves form a fit model.
+
+    .. math::
+
+        F(X, \Theta) = f_0(\vec{x}_0, \vec{p}_0) \oplus f_1(\vec{x}_1, \vec{p}_1) \oplus ...,
+
+    here the function :math:`f_i(\vec{x}_i, \vec{p}_i)` is applied to the data with the sequence
+    of x-values :math:`\vec{x}_i \in \Re^{N_i}`, which are provided by the :math:`i`-th subset
+    of experiments, :math:`E_i(\vec{x}_i) = \{E_i(x_0), E_i(x_1), ... E_i(x_{N_i-1})\}`,
+    together with the measured outcomes :math:`\vec{y}_i \in \Re^{N_i}` that are fit by this model.
+    The size of vector :math:`N_i` may depend on the configuration of experiment :math:`i`.
+
+    The parameter :math:`\vec{p}_i = \theta_i \cup \Theta_{\rm fix}` is a union of the
+    fit parameter for the function :math:`f_i` and the fixed parameters :math:`\Theta_{\rm fix}`.
+    The composite function :math:`F` consists of multiple fit functions :math:`f_i`
+    taking independent data points :math:`\vec{x}_i` with
+    partly shared fit parameters :math:`\theta_i`,
+    where :math:`\Theta = \theta_0 \cup \theta_1 \cup ...`.
+    The composite scan values is :math:`X =\vec{x}_0 \oplus \vec{x}_1 \oplus ...` and
+    the corresponding outcome is :math:`Y =\vec{y}_0 \oplus \vec{y}_1 \oplus ...`.
+
+    In the Qiskit Experiments, these data sources are represented by
+    a single flat array ``vec[k]``, rather than a jugged array ``mat[i, j]``.
```

Comment:
```suggestion
    a single 1D array ``vec[k]``, rather than a 2D array ``mat[i, j]``.
```
I don't know what `jugged` means in this context.

Reply:
It may take different element size for every `i`. Then this is not a clean 2D array of NxM.
https://en.wikipedia.org/wiki/Jagged_array